src/wallet/wallet.cpp

```diff
@@ -1474,6 +1474,34 @@ bool DummySignInput(const SigningProvider& provider, CTxIn &tx_in, const CTxOut
     return true;
 }
 
+static void FillInputToWeight(CTxIn& txin, int64_t target_weight)
```

Comment:
Please add some unit tests for this function? Given that there's so much consideration for boundaries, etc.

Reply:
Done